docs/Blog-Post.md

```diff
@@ -0,0 +1,525 @@
+still maintaining efficiency and scalability.
+
+Today, we're releasing the first version of Levanter and Haliax, along with several models trained with Levanter. We
+hope that these libraries will be useful to the community, and we look forward to seeing what people do with them.
+
+## The Landscape of Foundation Model Training Frameworks
+
+Numerous foundation model training frameworks exist in the community, each with its strengths and focuses.
+For large language models (LLMs) (the focus of this release),
+the most well-known in the open source community is probably NVIDIA's PyTorch-based [Megatron-LM](https://github.com/NVIDIA/Megatron-LM),
+and its many derivatives, including EleutherAI's [GPT-NeoX](https://github.com/EleutherAI/gpt-neox) codebase.
+Meta has [MetaSeq](https://github.com/facebookresearch/metaseq) as well as [FairScale](https://github.com/facebookresearch/fairscale),
+with which they trained [Llama](https://github.com/facebookresearch/llama).
+MosaicML has recently released [LLM Foundry](https://github.com/mosaicml/llm-foundry), which we used to train
+[BioMedLM](https://crfm.stanford.edu/2022/12/15/biomedlm.html) last autumn. Previously, we released [Mistral](
+https://github.com/stanford-crfm/mistral) built on [Hugging Face Transformers](https://github.com/huggingface/transformers/)
+and [DeepSpeed](https://github.com/microsoft/DeepSpeed).
+
+In the JAX community, there are a number of libraries popping up. Google has released [T5X](https://github.com/google-research/t5x)
+and [MaxText](https://github.com/google/maxtext). There are also a number of independent libraries, including [EasyLM](https://github.com/young-geng/EasyLM)
+and [JAXSeq](https://github.com/Sea-Snell/JAXSeq), both of which are based on [Flax](https://github.com/google/flax/)
+and modified libraries from [Hugging Face Transformers](https://github.com/huggingface/transformers/). Salesforce has
+released the Haiku-based [JAXformer](https://github.com/salesforce/jaxformer). Previously, Eleuther AI released
+[mesh-transformer-jax](https://github.com/kingoflolz/mesh-transformer-jax/), though it is mostly unmaintained now and uses
+older, quasi-deprecated JAX APIs for distributed training.
+
+## A New Codebase for Foundation Model Training
+
+Despite the wide array of existing frameworks, when we started, we found that none of them fully addressed our needs.
+At CRFM, we focused on three fundamental goals:
+
+* **Legibility and Composability**: We prioritized writing code that is easy to read, understand, and compose.
+* **Reproducibility and Resilience**: We emphasized the ability to reproduce results *exactly*, even in the face of preemption and restarts from checkpoints.
+* **Scalability and Efficiency**: We aimed to fully utilize the Cloud TPU resources from Google's TPU Research Cloud program, as well as our own GPU resources, without compromising efficiency.
```

Comment:
"Scalability"

Reply:
I'd focus on the levanter codebase, not the TPU research program (which is specific to our setting)